gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -102,9 +101,7 @@ public SeekableByteChannel position(long newPosition) throws IOException {
         return this;
       }
       position = newPosition;
-      size = fetchSize(gcsService, file);
-      channel.close();
-      channel = gcsService.openReadChannel(file, position);
+      channel.seek((int) newPosition);
```

Comment:
AFAIK this just fixed a code smell (do not update state if seek method throws an exception). This would only be relevant in an implementation where the underlying channel.seek method throws an exception instead of seeking, but our code already catches the documented preconditions that would lead to an exception (<0 target, closed channel). I see this more as defense in depth for when a maintainer comes along and goes "oh we don't need to call checkOpen here, channel.seek already does that check for us anyways".

I can't think of a reasonable handwritten test that would catch this.